src/tool/xmeta/cpp/models/namespace_model.cpp

```diff
@@ -0,0 +1,130 @@
+#include "namespace_model.h"
+
+#include "using_directive_models.h"
+
+#include "class_model.h"
+#include "delegate_model.h"
+#include "enum_model.h"
+#include "interface_model.h"
+#include "property_model.h"
+#include "struct_model.h"
+
+namespace xlang::xmeta
+{
+    void namespace_body_model::add_class(std::shared_ptr<class_model> const& cm)
+    {
+        m_classes[cm->get_id()] = cm;
```

Comment:
Use `insert` instead. This usage of the array subscript operator has two issues:
1. It will silently overwrite an existing entry. `insert`'s return value allows you to check if the insertion was blocked because of an existing element, which allows us to throw an error.
2. It's less performant for insertions. The array subscript operator constructs a `value_type` key-value pair, with the value half default constructed, and then immediately overwrites the value.

Reply:
Interesting, makes sense.